src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs

```diff
@@ -344,6 +405,159 @@ private void AddSingleCounterGroup(LazyStringValue docId, BlittableJsonReaderObj
                     _currentRun.AddCounter(docId, prop.Name, value);
             }
         }
+        
+        private void AddTimeSeries(LazyStringValue docId, Dictionary<string, IEnumerable<TimeSeriesStorage.Reader.SingleResult>> timeSeriesReaders)
+        {
+            foreach (var (timeSeriesName, entries) in timeSeriesReaders)
+            {
+                _currentRun.RemoveTimeSeries(docId, timeSeriesName, DateTime.MinValue, DateTime.MaxValue);
+                foreach (var singleResult in entries)
+                {
+                    _currentRun.AddTimeSeries(docId, timeSeriesName, singleResult);
+                }
+            }
+        }
+        
+        private void HandleSingleTimeSeriesSegment(LazyStringValue docId, TimeSeriesSegmentEntry segmentEntry, string function = null)
+        {
+            var timeSeriesEntries = segmentEntry.Segment.YieldAllValues(Context, segmentEntry.Start);
+            
+            if (function != null && FilterSingleTimeSeriesSegmentByScript(ref timeSeriesEntries, docId, segmentEntry, function))
+                return;
+
+            (DateTime from, DateTime to) toRemove = (default, default);
+            foreach (var entry in timeSeriesEntries)
```

Comment:
we are already iterating over `timeSeriesEntries ` in `FilterSingleTimeSeriesSegmentByScript` - can we handle removals of relevant entries as well?

Reply:
`FilterSingleTimeSeriesSegmentByScript` just define the filtretion by "iterator methods" the actual iteration is only here.

Reply:
OK, I see